### ISSUE TITLE
Collapse control component is not accessible

### ISSUE THREAD

**Issue by metamoni** [joemasilotti/railsdevs.com#553]
## Describe the bug
The `CollapseControlComponent` uses an expandable menu, but the state of the menu (expanded/collapsed) is not announced to screen readers.

## To reproduce

1. Go to the [developers page](https://railsdevs.com/developers)
2. Turn VoiceOver on: `command + F5`
3. Tab through the page until you get to the "Role level" filter
4. Notice VoiceOver does not announce "expanded"
5. Hit `Enter` to collapse the menu
6. Notice VoiceOver does not announce "collapsed"
7. Turn VoiceOver off: `command + F5`

## Expected behavior
When the menu is expanded and "Role level" is focused, VoiceOver should announce "expanded"
When the menu is collapsed and "Role level" is focused, VoiceOver should announce "collapsed"

## Actual behavior
VoiceOver does not announce "expanded" or "collapsed"

## Screenshots
_Current behavior_

https://user-images.githubusercontent.com/22390758/178143788-96979ae7-4ca3-4d75-8f05-2442ee9c3907.mov

_Example of expected behavior_

https://user-images.githubusercontent.com/22390758/178143798-5aa7f07a-491b-4458-ae32-d5e9226f94dd.mov

## Additional context
Use the `aria-expanded="true/false"` attribute in `app/components/collapse_control_component.html.erb` to dynamically update the state of the collapsible menu.
 
Further resources:
- [Collapsible sections by Heydon Pickering](https://inclusive-components.design/collapsible-sections/)
- Codepen: [an example of an accessible collapsible component](https://codepen.io/heydon/full/gGNaoM)

**Comment by atiqueakhtar** [comment]
hi @metamoni @joemasilotti , Can I please take up this issue and raise a PR to resolve it?

**Comment by joemasilotti** [comment]
Sure thing, all yours!

**Comment by joemasilotti** [comment]
Closed via #591.
